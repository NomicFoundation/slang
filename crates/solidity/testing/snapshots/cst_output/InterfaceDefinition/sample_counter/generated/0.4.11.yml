# This file is generated automatically by infrastructure scripts. Please don't edit by hand.

Source: >
  1  │ interface ICounter {                                                             │ 0..20
  2  │     // returns the current count                                                 │ 21..53
  3  │     function count() external view returns (uint);                               │ 54..104
  4  │                                                                                  │ 105..105
  5  │     // increments the counter                                                    │ 106..135
  6  │     function increment() external;                                               │ 136..170
  7  │ }                                                                                │ 171..172

Errors: []

Tree:
  - InterfaceDefinition (Rule): # 0..172 "interface ICounter {\n    // returns the current co..."
      - Interface (Token): "interface" # 0..9
      - Identifier (Token): "ICounter" # 10..18
      - _DELIMITEDBY (Rule): # 18..172 " {\n    // returns the current count\n    function c..."
          - OpenBrace (Token): "{" # 19..20
          - _REPEATED (Rule): # 21..171 "    // returns the current count\n    function coun..."
              - ContractBodyElement (Rule): # 21..106 "    // returns the current count\n    function coun..."
                  - FunctionDefinition (Rule): # 21..106 "    // returns the current count\n    function coun..."
                      - Function (Token): # 21..66 "    // returns the current count\n    function"
                          - SingleLineComment (Trivia): "// returns the current count" # 25..53
                          - Contents: "function" # 58..66
                      - Identifier (Token): "count" # 67..72
                      - ParameterList (Rule): # 72..74 "()"
                          - _DELIMITEDBY (Rule): # 72..74 "()"
                              - OpenParen (Token): "(" # 72..73
                              - CloseParen (Token): ")" # 73..74
                      - _REPEATED (Rule): # 75..88 "external view"
                          - FunctionAttribute (Rule): # 75..83 "external"
                              - External (Token): "external" # 75..83
                          - FunctionAttribute (Rule): # 84..88 "view"
                              - View (Token): "view" # 84..88
                      - Returns (Token): "returns" # 89..96
                      - ParameterList (Rule): # 96..103 " (uint)"
                          - _DELIMITEDBY (Rule): # 96..103 " (uint)"
                              - OpenParen (Token): "(" # 97..98
                              - _SEPARATEDBY (Rule): # 98..102 "uint"
                                  - ParameterDeclaration (Rule): # 98..102 "uint"
                                      - TypeName (Rule): # 98..102 "uint"
                                          - ElementaryType (Rule): # 98..102 "uint"
                                              - UnsignedIntegerType (Token): "uint" # 98..102
                              - CloseParen (Token): ")" # 102..103
                      - Semicolon (Token): ";" # 103..104
              - ContractBodyElement (Rule): # 106..171 "    // increments the counter\n    function increme..."
                  - FunctionDefinition (Rule): # 106..171 "    // increments the counter\n    function increme..."
                      - Function (Token): # 106..148 "    // increments the counter\n    function"
                          - SingleLineComment (Trivia): "// increments the counter" # 110..135
                          - Contents: "function" # 140..148
                      - Identifier (Token): "increment" # 149..158
                      - ParameterList (Rule): # 158..160 "()"
                          - _DELIMITEDBY (Rule): # 158..160 "()"
                              - OpenParen (Token): "(" # 158..159
                              - CloseParen (Token): ")" # 159..160
                      - _REPEATED (Rule): # 161..169 "external"
                          - FunctionAttribute (Rule): # 161..169 "external"
                              - External (Token): "external" # 161..169
                      - Semicolon (Token): ";" # 169..170
          - CloseBrace (Token): "}" # 171..172
