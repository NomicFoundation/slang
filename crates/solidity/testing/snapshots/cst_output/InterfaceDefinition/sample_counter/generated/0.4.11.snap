# This file is generated automatically by infrastructure scripts. Please don't edit by hand.

Errors: 0

█ Rule(InterfaceDefinition) - (0..172)                                   │
├──► Token(Interface) - (0..9)                                           │ interface
├──► Token(RawIdentifier) - (10..18)                                     │ ICounter
└──█ Rule(DelimitedContractBodyElements) - (19..172)                     │
   ├──► Token(OpenBrace) - (19..20)                                      │ {
   ├──█ Rule(ContractBodyElements) - (25..170)                           │
   │  ├──█ Rule(FunctionDefinition) - (25..104)                          │
   │  │  ├──█ Token(Function) - (25..66)                                 │
   │  │  │  ├──► Trivia(SingleLineComment) - (25..53)                    │ // returns the current count
   │  │  │  └──► Contents - (58..66)                                     │ function
   │  │  ├──► Token(RawIdentifier) - (67..72)                            │ count
   │  │  ├──█ Rule(ParameterList) - (72..74)                             │
   │  │  │  ├──► Token(OpenParen) - (72..73)                             │ (
   │  │  │  └──► Token(CloseParen) - (73..74)                            │ )
   │  │  ├──█ Rule(FunctionAttributes) - (75..88)                        │
   │  │  │  ├──► Token(External) - (75..83)                              │ external
   │  │  │  └──► Token(View) - (84..88)                                  │ view
   │  │  ├──█ Group - (89..103)                                          │
   │  │  │  ├──► Token(Returns) - (89..96)                               │ returns
   │  │  │  └──█ Rule(ParameterList) - (97..103)                         │
   │  │  │     ├──► Token(OpenParen) - (97..98)                          │ (
   │  │  │     ├──█ Rule(SeparatedParameterDeclarations) - (98..102)     │
   │  │  │     │  └──█ Rule(ParameterDeclaration) - (98..102)            │
   │  │  │     │     └──█ Rule(TypeName) - (98..102)                     │
   │  │  │     │        └──► Token(UnsignedIntegerType) - (98..102)      │ uint
   │  │  │     └──► Token(CloseParen) - (102..103)                       │ )
   │  │  └──► Token(Semicolon) - (103..104)                              │ ;
   │  └──█ Rule(FunctionDefinition) - (110..170)                         │
   │     ├──█ Token(Function) - (110..148)                               │
   │     │  ├──► Trivia(SingleLineComment) - (110..135)                  │ // increments the counter
   │     │  └──► Contents - (140..148)                                   │ function
   │     ├──► Token(RawIdentifier) - (149..158)                          │ increment
   │     ├──█ Rule(ParameterList) - (158..160)                           │
   │     │  ├──► Token(OpenParen) - (158..159)                           │ (
   │     │  └──► Token(CloseParen) - (159..160)                          │ )
   │     ├──█ Rule(FunctionAttributes) - (161..169)                      │
   │     │  └──► Token(External) - (161..169)                            │ external
   │     └──► Token(Semicolon) - (169..170)                              │ ;
   └──► Token(CloseBrace) - (171..172)                                   │ }
